src/spikeinterface/core/binaryrecordingextractor.py

```diff
@@ -156,25 +157,59 @@ def get_binary_description(self):
 class BinaryRecordingSegment(BaseRecordingSegment):
     def __init__(self, datfile, sampling_frequency, t_start, num_chan, dtype, time_axis, file_offset):
         BaseRecordingSegment.__init__(self, sampling_frequency=sampling_frequency, t_start=t_start)
-        self._timeseries = read_binary_recording(datfile, num_chan, dtype, time_axis, file_offset)
+        self.num_chan = num_chan
+        self.dtype = np.dtype(dtype)
+        self.file_offset = file_offset
+        self.time_axis = time_axis
+        self.datfile = datfile
+        self.file = open(self.datfile, "r")
+        self.num_samples = (Path(datfile).stat().st_size - file_offset) // (num_chan * np.dtype(dtype).itemsize)
+        if self.time_axis == 0:
+            self.shape = (self.num_samples, self.num_chan)
+        else:
+            self.shape = (self.num_chan, self.num_samples)
 
     def get_num_samples(self) -> int:
         """Returns the number of samples in this signal block
 
         Returns:
             SampleIndex: Number of samples in the signal block
         """
-        return self._timeseries.shape[0]
+        return self.num_samples
 
     def get_traces(
         self,
         start_frame: Union[int, None] = None,
         end_frame: Union[int, None] = None,
         channel_indices: Union[List, None] = None,
     ) -> np.ndarray:
-        traces = self._timeseries[start_frame:end_frame]
+        byte_offset = self.file_offset
+        dtype_size_bytes = self.dtype.itemsize
+        mmap_offset, array_offset = divmod(byte_offset, mmap.ALLOCATIONGRANULARITY)
+        data_size_bytes = dtype_size_bytes * self.num_samples * self.num_chan
+        mmmap_length = data_size_bytes + array_offset
```

Comment:
Just for the pleasure to be picky : this could be in the init, no ?

Reply:
Yep, makes sense, I moved it in the last commit.

That said, I think we could be even more efficient and only reserve the buffer that corresponds to the start_frame to end_frame. But let's do that in another PR.

Reply:
perfect!